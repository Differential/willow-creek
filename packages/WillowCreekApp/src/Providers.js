import { nest } from 'recompose';
import { Providers } from '@apollosproject/ui-kit';
import ClientProvider from './client';

export default nest(ClientProvider, Providers);
