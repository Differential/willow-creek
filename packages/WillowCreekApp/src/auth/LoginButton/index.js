import React, { PureComponent } from 'react';
import { withNavigation } from 'react-navigation';
import { Query } from 'react-apollo';
import PropTypes from 'prop-types';

import { Button } from '@apollosproject/ui-kit';

import getLoginState from '../getLoginState';

class LoginButton extends PureComponent {
  static propTypes = {
    navigation: PropTypes.shape({
      push: PropTypes.func,
    }),
  };

  handleLoginPress = () => this.props.navigation.push('Auth');

  render() {
    const { navigation, ...otherProps } = this.props;
    return (
      <Query query={getLoginState}>
        {({ data }) => {
          const { isLoggedIn, loading } = data;
          if (isLoggedIn) return null;
          return (
            <Button
              onPress={this.handleLoginPress}
              title="Get Connected"
              loading={loading}
              {...otherProps}
            />
          );
        }}
      </Query>
    );
  }
}

export default withNavigation(LoginButton);
