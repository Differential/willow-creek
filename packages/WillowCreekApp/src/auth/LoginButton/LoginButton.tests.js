import React from 'react';
import renderer from 'react-test-renderer';
import wait from 'waait';

import Providers from 'WillowCreekApp/src/Providers';

import getLoginState from '../getLoginState';

import LoginButton from '.';

describe('LoginButton component', () => {
  it('renders nothing when logged in', async () => {
    const mock = {
      request: {
        query: getLoginState,
      },
      result: {
        data: { isLoggedIn: true },
      },
    };

    const navigation = { navigate: jest.fn() };
    const tree = renderer.create(
      <Providers mocks={[mock]}>
        <LoginButton navigation={navigation} />
      </Providers>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });

  it('renders a LoginButton when logged out', async () => {
    const mock = {
      request: {
        query: getLoginState,
      },
      result: {
        data: { isLoggedIn: null },
      },
    };

    const navigation = { navigate: jest.fn() };
    const tree = renderer.create(
      <Providers mocks={[mock]}>
        <LoginButton navigation={navigation} />
      </Providers>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });

  it('renders a LoginButton that is loading', async () => {
    const mock = {
      request: {
        query: getLoginState,
      },
      result: {
        data: { isLoggedIn: null },
      },
    };

    const navigation = { navigate: jest.fn() };
    const tree = renderer.create(
      <Providers mocks={[mock]}>
        <LoginButton navigation={navigation} loading />
      </Providers>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });
});
