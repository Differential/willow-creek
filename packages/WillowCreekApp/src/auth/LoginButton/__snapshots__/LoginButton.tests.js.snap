// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton component renders a LoginButton that is loading 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        bordered={false}
        disabled={false}
        pill={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00676D",
            "borderColor": "#00676D",
            "borderRadius": 48,
            "borderWidth": 2,
            "elevation": 2,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "center",
            "opacity": 1,
            "overflow": "hidden",
            "paddingHorizontal": 16,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#FFFFFF"
          hidesWhenStopped={true}
          size="small"
        />
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`LoginButton component renders a LoginButton when logged out 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        bordered={false}
        disabled={false}
        pill={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00676D",
            "borderColor": "#00676D",
            "borderRadius": 48,
            "borderWidth": 2,
            "elevation": 2,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "center",
            "opacity": 1,
            "overflow": "hidden",
            "paddingHorizontal": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "InterUI-Bold",
              "fontSize": 16,
              "lineHeight": 23.04,
            }
          }
        >
          Get Connected
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`LoginButton component renders nothing when logged in 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</RCTSafeAreaView>
`;
